apps/src/pythonlab/pyodideWebWorker.js

```diff
@@ -35,15 +35,23 @@ async function loadPyodideAndPackages() {
   });
   self.pyodide.setStdout(getStreamHandlerOptions('sysout'));
   self.pyodide.setStderr(getStreamHandlerOptions('syserr'));
+  // Warm up the pyodide environment by running setup code.
+  await runInternalCode(SETUP_CODE, 'setup_run');
 }
 
 let pyodideReadyPromise = null;
 let pyodideGlobals = null;
 async function initializePyodide() {
+  let promiseWasNull = false;
   if (pyodideReadyPromise === null) {
     pyodideReadyPromise = loadPyodideAndPackages();
+    self.postMessage({type: 'loading_pyodide'});
+    promiseWasNull = true;
```

Comment:
unimportant nit, 100% could wontfix, what about:
```
const promiseWasNull = pyodideReadyPromise === null
if (promiseWasNull) {
  pyodideReadyPromise = loadPyodideAndPackages();
  self.postMessage({type: 'loading_pyodide'});
}
```

One line of code shorter, and uses one less mutable variable.

Reply:
updated!